thorium/resources.py

```diff
@@ -231,6 +234,8 @@ def validate(self):
 
     def _validate_full(self):
         for name, field in self._fields.items():
+            if not self.is_set(name) and not field.is_mutable:
+                self.to_default(name)
```

Comment:
If there's no default, the value should be `NotSet` and be caught in the very next if condition raising a `ValidationError`.